java/group-layers/src/main/java/com/esri/arcgisruntime/samples/grouplayers/MainActivity.java

```diff
@@ -0,0 +1,381 @@
+    groupLayer.getLayers().addAll(Arrays.asList(devOne, devTwo, devThree));
+
+    // add the group layer and other layers to the scene as operational layers
+    scene.getOperationalLayers().addAll(Arrays.asList(groupLayer, nonDevOne, nonDevTwo));
+
+    // zoom to the extent of the group layer when the child layers are loaded
+    ListenableList<Layer> layers = groupLayer.getLayers();
+    for (Layer childLayer : layers) {
+      childLayer.addDoneLoadingListener(() -> {
+        if (childLayer.getLoadStatus() == LoadStatus.LOADED) {
+          mSceneView.setViewpointCamera(new Camera(groupLayer.getFullExtent().getCenter(), 700, 0, 60, 0));
+        }
+      });
+    }
+
+    setupRecyclerView(scene.getOperationalLayers());
+  }
+
+  /**
+   * Setup {@link RecyclerView} to display layers
+   *
+   * @param layers
+   */
+  private void setupRecyclerView(List<Layer> layers) {
+    mLayersAdapter = new LayersAdapter(this);
+    mLayersRecyclerView.setAdapter(mLayersAdapter);
+
+    for (Layer layer : layers) {
+      // if layer can be shown in legend
+      if (layer.canShowInLegend()) {
+        layer.addDoneLoadingListener(() -> {
+          mLayersAdapter.addLayer(layer);
+        });
+        layer.loadAsync();
+      }
+    }
+  }
+
+  @Override public boolean onCreateOptionsMenu(Menu menu) {
+    getMenuInflater().inflate(R.menu.menu_main_activity, menu);
+    return true;
+  }
+
+  @Override public boolean onOptionsItemSelected(MenuItem item) {
+    if (item.getItemId() == R.id.action_show_layer_list) {
+      // if bottom sheet is not shown
+      if (mBottomSheetBehavior.getState() != BottomSheetBehavior.STATE_EXPANDED) {
+        // show bottom sheet
+        mBottomSheetBehavior.setState(BottomSheetBehavior.STATE_EXPANDED);
+      } else {
+        // hide bottom sheet
+        mBottomSheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN);
+      }
+      return true;
+    }
+    return super.onOptionsItemSelected(item);
+  }
+
+  /**
+   * Called when a checkbox on a layer in the list is checked or unchecked
+   *
+   * @param layer   that has been checked or unchecked
+   * @param checked whether the checkbox has been checked or unchecked
+   */
+  @Override public void layerCheckedChanged(Layer layer, boolean checked) {
+    layer.setVisible(checked);
+    if (layer instanceof GroupLayer) {
+      for (Layer childLayer : ((GroupLayer) layer).getLayers()) {
+        childLayer.setVisible(checked);
+      }
+    }
+  }
+
+  @Override
+  protected void onPause() {
+    mSceneView.pause();
+    super.onPause();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mSceneView.resume();
+  }
+
+  @Override
+  protected void onDestroy() {
+    mSceneView.dispose();
+    super.onDestroy();
+  }
+
+  /**
+   * An adapter that displays {@link Layer}s and their children if they are an instance of {@link GroupLayer}
+   */
+  private class LayersAdapter extends RecyclerView.Adapter<LayersAdapter.ViewHolder>
```

Comment:
Would it be alot of trouble to move the subclasses out of MainActivity? If not, could be good to move UI code out of MainActivity to increase visibility of Runtime API

Reply:
No problem.